compiler-rt/lib/builtins/fp_extend.h

```diff
@@ -20,14 +20,21 @@
 typedef float src_t;
 typedef uint32_t src_rep_t;
 #define SRC_REP_C UINT32_C
-static const int srcSigBits = 23;
+static const int srcBits = sizeof(src_t) * CHAR_BIT;
+static const int srcSigFracBits = 23;
```

Comment:
These constants are already defined inside fp_lib.h. Wouldn't it make more sense to refactor fp_lib.h so that instead of 
```
#if defined SINGLE_PRECISION
...
#define significandBits 23
#elif defined DOUBLE_PRECISION
...
#define significandBits 52
```
we have something like
```
#define sf_significand_bits 23
#define df_significand_bits 52
#define xf_significand_bits 63
#if defined SINGLE_PRECISION
#define significandBits sf_significand_bits
#elif defined DOUBLE_PRECISION
#define significandBits df_significand_bits
```
and then reuse those constants here?

Reply:
It definitely would, but I wanted to apply these changes in multiple phases 
(see also the description)  (partially to limit the scope of this PR and partially to avoid concurrently editing fp_lib.h (so that multiple PRs could make progress/land independently / a bit easier).